commands/project_doctor.go

```diff
@@ -0,0 +1,213 @@
+	fmt.Println("Project doctor evaluates project-specific environment issues.")
+	fmt.Println("You will find most of the checks defined in your Outrigger Project configuration (e.g., outrigger.yml)")
+	fmt.Println("These checks are not comprehensive, this is intended to automate common environment troubleshooting steps.")
+	fmt.Println()
+	compendium, _ := cmd.GetConditionCollection()
+	if err := cmd.AnalyzeConditionList(compendium); err != nil {
+		// Directly returning the framework error to skip the expanded help.
+		// A failing state is self-descriptive.
+		return cli.NewExitError(fmt.Sprintf("%v", err), 1)
+	}
+
+	return nil
+}
+
+// RunCompendium lists all conditions to be checked in the analysis.
+func (cmd *ProjectDoctor) RunCompendium(ctx *cli.Context) error {
+	compendium, _ := cmd.GetConditionCollection()
+	cmd.out.Info("There are %d conditions in the repertoire.", len(compendium))
+	fmt.Println(compendium)
+
+	return nil
+}
+
+// AnalyzeConditionList checks each registered condition against environment state.
+func (cmd *ProjectDoctor) AnalyzeConditionList(conditions ConditionCollection) error {
+	var returnVal error
+
+	failing := ConditionCollection{}
+	for _, condition := range conditions {
+		cmd.out.Spin(fmt.Sprintf("Examining project environment for %s", condition.Name))
+		if found := cmd.Analyze(condition); !found {
+			cmd.out.Info("Not Affected by: %s [%s]", condition.Name, condition.Id)
+		} else {
+			switch condition.Severity {
+			case ConditionSeverityWARNING:
+				cmd.out.Warning("Condition Detected: %s [%s]", condition.Name, condition.Id)
+				failing[condition.Id] = condition
+				break
+			case ConditionSeverityERROR:
+				cmd.out.Error("Condition Detected: %s [%s]", condition.Name, condition.Id)
+				failing[condition.Id] = condition
+				if returnVal == nil {
+					returnVal = errors.New("Diagnosis found at least one failing condition.")
+				}
+				break
+			default:
+				cmd.out.Info("Condition Detected: %s [%s]", condition.Name, condition.Id)
+			}
+		}
+	}
+
+	if len(failing) > 0 {
+		color.Red("\nThere were %d problems identified out of %d checked.\n", len(failing), len(conditions))
+		fmt.Println(failing)
+	}
+
+	return returnVal
+}
+
+// GetConditionCollection assembles a list of all conditions.
+func (cmd *ProjectDoctor) GetConditionCollection() (ConditionCollection, error) {
+	conditions := cmd.Config.Doctor
+
+	// @TODO move these to outrigger.yml once we have pure shell facilities.
+	eval := ProjectEval{cmd.out, cmd.Config}
+	sync := ProjectSync{}
+	syncName := sync.GetVolumeName(cmd.Config, eval.GetWorkingDirectory())
+
+	// @todo we should have a way to determine if the project wants to use sync.
+	item1 := &Condition{
+		Id:           "sync-container-not-running",
+		Name:         "Sync Container Not Working",
+		Test:         []string{fmt.Sprintf("$(id=$(docker container ps -aq --filter 'name=^/%s$'); docker top $id &>/dev/null)", syncName)},
```

Comment:
Some potential shell versions assuming you are ok with the sync volume being hard coded in outrigger.yml.

`test -z $(docker container ps -aq --filter 'name=^/projectname-sync$')` should tell you if the container exists but not much about if it is running.

`test -z $(docker container ps -q --filter 'name=^/projectname-sync$')` dropping the -a flag should let you know if it is running but can't tell you anything about the processes inside it. I don't think that is different than what the return code from docker top is doing for you but I'm not positive.

Reply:
The problem is `docker container ps` never has a failing exit code, whereas `docker top` does.

After creating these examples of conditions, I've since gone on to work on have a more hard-coded set of checks around project sync, I hope to have a PR ready later today/tomorrow.

With all the issues around unison, I want to be able to tell people to upgrade rig and have a comprehensive solution in place for all projects using unison.